.github/workflows/build.yml

```
@@ -0,0 +1,122 @@
+name: Build CI
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    branches: ["*"]
+  workflow_dispatch:
+
+env:
+  BUILD_CONFIG: RelWithDebInfo
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - shell_arch: x64
+            win_arch: x64
+          - shell_arch: x86
+            win_arch: x86
+          - shell_arch: amd64_arm64
+            win_arch: arm64
+          - shell_arch: amd64_arm
+            win_arch: arm
+
+    name: Build for ${{ matrix.win_arch }}
+    runs-on: windows-latest
+
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+      
+    - name: Setup shell
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ matrix.shell_arch }}
+    
+    - name: Prepare build
+      run: cmake -B build -G "Ninja"
+      
+    - name: Build
+      run: cmake --build build --config ${{env.BUILD_CONFIG}} -j 3
+      
+    - name: Prepare files for upload
+      run: |
+        mkdir upload
+        mkdir upload/${{ matrix.win_arch }}
+
+        copy src/btrfs.inf upload
+        copy build/ubtrfs.dll upload/${{ matrix.win_arch }}
+        copy build/shellbtrfs.dll upload/${{ matrix.win_arch }}
+        copy build/mkbtrfs.exe upload/${{ matrix.win_arch }}
+        copy build/btrfs.sys upload/${{ matrix.win_arch }}
+
+        mkdir upload-pdb
+        mkdir upload-pdb/${{ matrix.win_arch }}
+
+        copy build/ubtrfs.pdb upload-pdb/${{ matrix.win_arch }}
+        copy build/shellbtrfs.pdb upload-pdb/${{ matrix.win_arch }}
+        copy build/mkbtrfs.pdb upload-pdb/${{ matrix.win_arch }}
+        copy build/btrfs.pdb upload-pdb/${{ matrix.win_arch }}
```

```suggestion
        '', "/${{ matrix.win_arch }}", '-pdb', "-pdb/${{ matrix.win_arch }}" | ForEach-Object { New-Item -ItemType Directory "upload$_" }
        
        '', '-vol' | ForEach-Object { Copy-Item "src/$_.inf" upload }
        
        'ubtrfs.dll', 'shellbtrfs.dll', 'mkbtrfs.exe', 'btrfs.sys'  | ForEach-Object {
            Copy-Item $_ upload/${{ matrix.win_arch }}
            Copy-Item ($_.Substring(0, $_.Length - 3) + 'pdb') upload-pdb/${{ matrix.win_arch }}
        }
```
Should be PowerShell (`mkdir` is not PowerShell), and should copy `btrfs-vol.inf` as well

---

If your interest in this is because you are planning to run your own fork with different filename limits, do NOT do this. You will corrupt your filesystem when you use it on Linux, as even with the changes on that Russian site the kernel driver will overwrite other data with the long filename.

---

Nah that's not my sole reason, I did try that. It didn't corrupt anything, but I did get a warning on Linux, and renaming such paths removes any warnings, and just works.